guide/src/reference/types/enum.md

```diff
@@ -0,0 +1,72 @@
+# enum
+
+| `T` parameter | `&T` parameter | `&mut T` parameter | `T` return value | `Option<T>` parameter | `Option<T>` return value | JavaScript representation |
+| :-----------: | :------------: | :----------------: | :--------------: | :-------------------: | :----------------------: | :-----------------------: |
+|      Yes      |       No       |         No         |        No        |          Yes          |           Yes            |   `string` or `number`    |
+
+## Example Rust Usage
+
+```rust
+{{#include ../../../../examples/guide-supported-types-examples/src/enum.rs}}
+```
+
+## Example JavaScript Usage
+
+```js
+{{#include ../../../../examples/guide-supported-types-examples/enum.js}}
+```
+
+## TypeScript
+
+Unfortunately, string enums don't fully work yet; no TypeScript is generated for the enum itself and functions using them accept or return `any`.
+They work correctly, it's just there's no type hints.
+See [Issue #3057](https://github.com/rustwasm/wasm-bindgen/issues/3057)
```

Comment:
```suggestion
Note that string enums currently don't generate a type and instead use `any`.
```
Let's keep the language neutral like in the rest of the documentation.
Also I don't believe we link to issues in the rest of the documentation as well.

Reply:
I decided to link it so it doesn't go out of date. i agree on the wording change.

Reply:
like if string enums get fixed and the documentation doesn't get updated, hopefully someone will open the issue, see that it's completed, and realize "oh, the documentation must be outdated"